src/components/stepforms/ConcatenateStepForm.vue

```diff
@@ -0,0 +1,92 @@
+<template>
+  <div>
+    <step-form-title :title="title"></step-form-title>
+    <ListWidget
+      addFieldName="Add columns"
+      id="toConcatenate"
+      name="Columns to concatenate:"
+      v-model="toConcatenate"
+      :defaultItem="[]"
+      :widget="columnPicker"
+      :componentProps="{sync: false}"
+      :automatic-new-field="false"
+      data-path=".columns"
+      :errors="errors"
+    ></ListWidget>
+    <InputTextWidget
+      id="separator"
+      v-model="editedStep.separator"
+      name="Separator:"
+      placeholder="Enter string of any length"
+      data-path=".separator"
+      :errors="errors"
+    ></InputTextWidget>
+    <InputTextWidget
+      id="newColumnName"
+      v-model="editedStep.new_column_name"
+      name="New column name:"
+      placeholder="Enter a columnn name"
+      data-path=".new_column_name"
+      :errors="errors"
+    ></InputTextWidget>
+    <step-form-buttonbar :cancel="cancelEdition" :submit="submit"></step-form-buttonbar>
+  </div>
+</template>
+
+<script lang="ts">
+import { Prop } from 'vue-property-decorator';
+import { StepFormComponent } from '@/components/formlib';
+import ColumnPicker from '@/components/stepforms/ColumnPicker.vue';
+import InputTextWidget from '@/components/stepforms/widgets/InputText.vue';
+import ListWidget from '@/components/stepforms/widgets/List.vue';
+import BaseStepForm from './StepForm.vue';
+import { ConcatenateStep } from '@/lib/steps';
+
+@StepFormComponent({
+  vqbstep: 'concatenate',
+  name: 'concatenate-step-form',
+  components: {
+    ColumnPicker,
+    InputTextWidget,
+    ListWidget,
+  },
+})
+export default class ConcatenateStepForm extends BaseStepForm<ConcatenateStep> {
+  @Prop({
+    type: Object,
+    default: () => ({ name: 'concatenate', columns: [''], separator: '', new_column_name: '' }),
+  })
+  initialStepValue!: ConcatenateStep;
+
+  readonly title: string = 'Concatenate columns';
+  columnPicker = ColumnPicker;
```

Comment:
Should that be a prop?

Reply:
It conforms to how is built every other component with repeatable widget.
I understand it would change the pattern of how we instantiate components with such a property then?
Maybe it's a change that can e proposed in a specific PR?